topology/graph/client.go

```diff
@@ -169,8 +173,12 @@ func (c *AsyncClient) AddListener(l EventListener) {
 
 func (c *AsyncClient) Disconnect() {
 	c.running.Store(false)
-	c.quit <- true
+	if c.connected.Load() == true {
+		c.wsConn.Close()
+		c.connected.Store(false)
```

Comment:
> But if the connection is not established, you should not call wsconn.Close()
If the connection is not established "connected" is not set to true so no Close

> the if running line 146 make no sense anymore, as close(c.quit) will send a broadcast and the loop return directly
the running line 146 is needed as we can have a close from the line 177 and we need to know if already closed or not. Close + running false meaning do not try to connect anymore.

> o close(c.quit) need to be called just after c.running.Store(false)
no as if we don't wait that the goroutine end to close the chan the goroutine will write on a closed chan

I tested it a didn't any issue. Do we have an issue anymore ?